test-things/design-system
# Add HTML/Jinja Tabs to component page

## Additions

- adds HTML and Jinja Tabs markup and JS for viewing the code snippets
- adds field Jinja code snippet. It's a markdown field so we can also add links to Jinja templates in cfgov-refresh, for components where it doesn't make sense to copy/paste in Jinja code (such as really long, complex components). Code can still be in this field.

## Removals

- 

## Changes

- 

## Testing

1. check out https://deploy-preview-34--test-design-system.netlify.com/design-system/components/buttons and click the html and jinja tabs for each variation
1. click the show/hide code toggle button to make sure all the code & specs are hidden or shown for each variation on each component

## Screenshots


## Notes

what i had to do to use gov.uk's Tabs module:

### JS
Steps from here https://github.com/alphagov/govuk-frontend/blob/master/docs/installation/installing-with-npm.md
- install package with yarn: `yarn add govuk-frontend`
- in our project's `main.js`, import JS for Tabs module from node_modules
- add 'js-enabled' class to body element (? ) with  modernizr??
  - no, we have a no-js default class on html and then js removes it. 
  - i need to add no-js scoped styles that make hidden elements visible.
  - instead of rewriting all the CSS, I add a `js-enabled class` to the main element on page load to make the tab styles work. we can revisit this when we fork and contribute the Tab as a component to the design system.

### CSS
they don't provide compiled CSS for each component, only for the entire library. More info at https://github.com/alphagov/govuk-frontend/blob/master/docs/installation/installing-with-npm.md#importing-styles
- install sass
- compile the gov.uk .scss file node_modules to our own CSS file:
sass node_modules/govuk-frontend/components/tabs/_tabs.scss assets/css/tabs.less
- save CSS file to our project

### HTML
I copied and pasted from here https://design-system.service.gov.uk/components/tabs/

## Todos

- what do we do when there is no jinja code? should we even have tabs in that instance? will there be ever be variations that have html and no jinja?
- designer input is needed to style the tabs, CSS will need edits to match our own design standards, since this is a wireframe i'm leaving it as is for now
- at some point once we have design, styles and a design-system repo let's port the Tabs code to a Real Component in our system.

## Checklist


## Testing checklist

### Browsers


### Accessibility


### Other
